brave/common/tor/tor.mojom

```diff
@@ -0,0 +1,16 @@
+// Copyright 2018 The Brave Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+module tor.mojom;
+
+import "mojo/common/file_path.mojom";
+
+const string kTorServiceName = "tor_launcher";
+
+interface TorLauncher {
+  Launch(mojo.common.mojom.FilePath tor_bin, array<string> args)
+    => (bool result);
+
+  IsTorAlive() => (bool result);
```

Comment:
Does anything use this?  Seems to me we really want a notification for when tor is ready and when tor dies, rather than a way to poll whether tor is currently alive.

Reply:
Currently no consumers for this API. We can break the message pipe when tor dies and `BraveBrowserContext::OnTorLauncherCrashed` will be triggered. 
For tor is ready, `BraveBrowserContext::OnTorLaunched` is called right after fork/exec and pass the result.

Reply:
Or we can register a callback and call it when tor process dies